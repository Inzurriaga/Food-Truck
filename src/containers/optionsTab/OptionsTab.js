import React, { Component } from "react"
import { updateTrucksArray } from "../../thunk/UpdateTrucksArray"
import { connect } from "react-redux"

export class OptionsTab extends Component{
    constructor(){
        super();
        this.state = {
            location: "",
            distance: 1,
            open: false,
        }
    }

    changeState = (e) => {
        const { value, className } = e.target
        this.setState({
            [className]: value
        })
    }

    openTab = (e) => {
        e.preventDefault()
        this.setState({
            open: !this.state.open
        })
    }

    submitSettings = (e) => {
        const { distance, location } = this.state
        this.openTab(e)
        this.props.updateTrucksArray(distance, location)
        e.preventDefault()
    }

    render(){
        let tab;
        let arrow;
        if(this.state.open){
            tab = {
                left: "0%"
            }
            arrow = {
                transform: "rotate(180deg)"
            }
        }else{
            tab = {
                left: "-22%"
            }
            arrow = {
                transform: "rotate(0deg)"
            }
        }
        return(
            <form style={tab} className="option-tab">
                <div style={arrow} onClick={this.openTab} className="open-button"></div>
                <h2>Settings</h2>
                <section className="location">
                    <h3>Location</h3>
                    <input onChange={this.changeState} className="location" value={this.state.location}></input>
                </section>
                <section>
                    <h3>Distance</h3>
                    <div className="select">
                        <select className="distance" onChange={this.changeState}>
                            <option value="1">1 mile</option>
                            <option value="5">5 mile</option>
                            <option value="10">10 mile</option>
                            <option value="25">25 mile</option>
                        </select>
                    </div>
                </section>
                <button onClick={this.submitSettings} className="submit">submit</button>
            </form>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    updateTrucksArray: (distance, url) => dispatch(updateTrucksArray(distance, url))
})

export default connect(null, mapDispatchToProps)(OptionsTab)