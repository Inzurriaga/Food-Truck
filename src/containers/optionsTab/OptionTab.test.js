import React from "react";
import { shallow } from "enzyme";
import OptionsTab from "./OptionsTab"

describe("OptionsTab", () => {
    describe("OptionsTab", () => {
        let wrapper;
        beforeEach(() => {
            wrapper = shallow(
                <OptionsTab />
            )
        })
        it("should match snap shot", () => {
            expect(wrapper).toMatchSnapshot()
        })
    })
})