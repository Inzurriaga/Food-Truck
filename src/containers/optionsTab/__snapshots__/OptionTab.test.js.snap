// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsTab OptionsTab should match snap shot 1`] = `
<form
  className="option-tab"
  style={
    Object {
      "left": "-22%",
    }
  }
>
  <div
    className="open-button"
    onClick={[Function]}
    style={
      Object {
        "transform": "rotate(0deg)",
      }
    }
  />
  <h2>
    Settings
  </h2>
  <section
    className="location"
  >
    <h3>
      Location
    </h3>
    <input
      className="location"
      onChange={[Function]}
      value=""
    />
  </section>
  <section>
    <h3>
      Distance
    </h3>
    <div
      className="select"
    >
      <select
        className="distance"
        onChange={[Function]}
      >
        <option
          value="1"
        >
          1 mile
        </option>
        <option
          value="5"
        >
          5 mile
        </option>
        <option
          value="10"
        >
          10 mile
        </option>
        <option
          value="25"
        >
          25 mile
        </option>
      </select>
    </div>
  </section>
  <button
    className="submit"
    onClick={[Function]}
  >
    submit
  </button>
</form>
`;
